flake8_rst/rst.py

```diff
@@ -20,15 +19,51 @@
 )
 
 ANCHOR_RE = re.compile(
+    r'(?P<before>.*\n)'
+    r'(?P<code>^>{3} (.+\n)+)',
+    re.MULTILINE,
+)
+
+INTERN_ANCHOR_RE = re.compile(
+    r'(?P<before>\n*)'
+    r'^((?P<indent>(?:>{3}|\.{3}) ?)(?P<code>.*))$|^(?!>{3}|\.{3})(?P<output>.*)$',
+    re.MULTILINE,
+)
+
+ANCHOR_IN_RE = re.compile(
     r'(?P<before>'
-    r'(?P<code>('
-    r'^(?P<indent> *)>>> .*\n'
-    r'^(((?P=indent)(>>>|...)(.*))?\n)+)'
-    r'))',
+    r'\n*'
+    r')'
+    r'^(?P<indent>(?:(?:In \[(\d+)\]:)|(?: +\.{4}:)) ?)(?P<code>.*)$|^(?!In \[(\d+)\]:)|(?: +\.{4}:)(?P<output>.*)$',
     re.MULTILINE,
 )
 
-EXPRESSIONS = (RST_RE, ANCHOR_RE)
+EXPRESSIONS = [RST_RE, ANCHOR_RE]
+
+
+def strip_pycon(src, indent, line_number, *expressions):
+    has_stripped = False
+    for code_block in src.split('\n\n'):
+        matches = (match for expression in expressions for match in re.finditer(expression, code_block))
+        current_block = []
+        for match in matches:
+            origin_code = match.group('code')
+            indent_ = match.group('indent')
+            output = match.group('output')
+
+            if origin_code is not None:
+                current_block.append((origin_code, len(indent_) if indent_ else 0))
```

Comment:
Maybe make `indent_ = match.group('indent') or ''` and leave here only `len(indent_)`.
Also why you add _ to indent variable name?